test/integration/Resends.test.js

```diff
@@ -0,0 +1,170 @@
+import { ethers } from 'ethers'
+import uuid from 'uuid/v4'
+
+import StreamrClient from '../../src'
+
+import config from './config'
+
+const { wait, waitForCondition } = require('streamr-test-utils')
+
+const createClient = (opts = {}) => new StreamrClient({
+    url: config.websocketUrl,
+    restUrl: config.restUrl,
+    auth: {
+        privateKey: ethers.Wallet.createRandom().privateKey,
+    },
+    autoConnect: false,
+    autoDisconnect: false,
+    ...opts,
+})
+
+const MAX_MESSAGES = 10
+const TEST_REPEATS = 10
+
+describe('StreamrClient resends', () => {
+    describe('resend', () => {
+        let client
+        let stream
+        let publishedMessages
+
+        beforeEach(async () => {
+            client = createClient()
+            await client.ensureConnected()
+
+            publishedMessages = []
+
+            stream = await client.createStream({
+                name: uuid(),
+            })
+
+            for (let i = 0; i < MAX_MESSAGES; i++) {
+                const message = {
+                    msg: `message${i}`,
+                }
+
+                // eslint-disable-next-line no-await-in-loop
+                await client.publish(stream.id, message)
+                publishedMessages.push(message)
+            }
+
+            await wait(3000) // wait for messages to (hopefully) land in storage
+        }, 10 * 1000)
+
+        afterEach(async () => {
+            await client.ensureDisconnected()
+        })
+
+        it('resend last using resend function', async () => {
+            for (let i = 0; i < TEST_REPEATS; i++) {
+                const receivedMessages = []
+
+                // eslint-disable-next-line no-await-in-loop
+                const sub = await client.resend(
+                    {
+                        stream: stream.id,
+                        resend: {
+                            last: MAX_MESSAGES,
+                        },
+                    },
+                    (message) => {
+                        receivedMessages.push(message)
+                    },
+                )
+
+                // eslint-disable-next-line no-loop-func
+                sub.once('resent', () => {
+                    expect(receivedMessages).toStrictEqual(publishedMessages)
+                })
+
+                // eslint-disable-next-line no-await-in-loop
+                await waitForCondition(() => receivedMessages.length === MAX_MESSAGES, 10000)
+            }
+        }, 50000)
+
+        it('resend last using subscribe function', async () => {
+            for (let i = 0; i < TEST_REPEATS; i++) {
```

Comment:
one alternative way of achieving this is to do something like this: 

https://github.com/streamr-dev/engine-and-editor/pull/707/files#diff-71db015fd49f36fd3008ea2fdb12148dR315-R395

i.e. put the `it` inside the `for` loop.

This has the benefit of telling you which repeat failed and while also allowing you to set a more normal timeout for each individual test, which should allow repeats to fail faster. e.g. in the example above the test would always pass on the first 2-3 times, but would fail sporadically after that. This was useful debugging information.

Just a suggestion.